### PR TITLE
Feature 163

### PR DESCRIPTION
**Pull request by meronbrouwer**
Hi @AbdelHajou, I don't know if you can make the time, but this pull request adds the option to nullify the speed of an Entity in a specific direction. Its the behaviour you need to make an entity collide with something and then stop its motion in the direction of the colliding object (imagine a fly that collides with a wall).

I have also added some JavaDoc for this one, that uses both svg images for clarity and a LaTex based javascript math formula parsing formula.

### CODE REVIEW

File: src/main/java/com/github/hanyaeger/api/engine/entities/entity/motion/MotionModifier.java [hunk 1]
```diff
@@ -76,6 +76,32 @@
...
+
+    /**
+     * Since the motion can be described as a vector, such a vector can be decomposed in two perpendicular
+     * components. Calling this methods negates one of those two components, leaving the other as the new motion.
+     *
+     * <p>
+     * <img width="30%" src="doc-files/negate-motion-vectors.svg" alt="Vector representation of negating the motion in a given direction">
+     *
+     * <p> In the image above, let <b>v</b> denote the vector associated with the current motion and <b>b</b> be the vector
+     * in the direction that has to be negated, then the resulting vector <b>p</b> van be calculated by
```
Comment (AbdelHajou) [inline src/main/java/com/github/hanyaeger/api/engine/entities/entity/motion/MotionModifier.java:93]: van -> can

File: src/main/java/com/github/hanyaeger/api/engine/entities/entity/motion/MotionModifier.java [hunk 2]
```diff
@@ -76,6 +76,32 @@
...
+     * <p>
+     * \( \textbf{p} = \cfrac{\textbf{v}\cdot\textbf{b}}{\textbf{b}\cdot\textbf{b}} \cdot \textbf{b}   \)
+     *
+     * <p>A typical use case would be an entity that jumps with a parabolic motion. At one point it collides with
+     * the ground, which should cancel the vertical downward motion. The vertical motion, however, should not be
+     * canceled. In this case, one should call {@code negateMotionInDirection(Direction.DOWN)}.
```
Comment (AbdelHajou) [inline src/main/java/com/github/hanyaeger/api/engine/entities/entity/motion/MotionModifier.java:99]: canceled -> cancelled

**Review by AbdelHajou — APPROVED** [review]
Looks good, there's just 2 typos in the Javadoc.

**Comment by AbdelHajou** [comment]
Do you know any IntelliJ plugins that can render the javadoc with images and LaTeX in the IDE?

**Comment by meronbrouwer** [comment]
@AbdelHajou Thanks, I'll make the changes. You can just run "mvn javadoc:javadoc", which creates the html in the target/ folder

File: src/main/java/com/github/hanyaeger/api/engine/entities/entity/motion/MotionModifier.java (hunk 1 above)
Comment (meronbrouwer) [reply src/main/java/com/github/hanyaeger/api/engine/entities/entity/motion/MotionModifier.java]: fixed

File: src/main/java/com/github/hanyaeger/api/engine/entities/entity/motion/MotionModifier.java (hunk 2 above)
Comment (meronbrouwer) [reply src/main/java/com/github/hanyaeger/api/engine/entities/entity/motion/MotionModifier.java]: fixed

**Pull request merged**